src/HttpClient.ts

```diff
@@ -201,7 +201,8 @@ export class HttpClient extends EventEmitter {
         // url maybe url.parse(url) object in urllib2
         requestUrl = new URL(urlFormat(url));
       } else {
-        requestUrl = url;
+        // Don't mutate the URL object the user passed in
+        requestUrl = new URL(url);
```

Comment:
should change url to a string before pass to URL 

```ts
requestUrl = new URL(url.toString());
```

Reply:
The URL constructor does that automatically; it's perfectly happy being passed a URL object or anything else with a `toString` method.

Reply:
Oh, weird, the node:url URL type wants a string, but the whatwg URL type (in browsers and with types provided by the dom module of typescript) is happy to take another URL. I'll add the .toString(), and a test.